source/binary.cpp

```diff
@@ -732,6 +739,41 @@ spv_result_t Parser::setNumericTypeInfoForType(
   return SPV_SUCCESS;
 }
 
+spv_result_t Parser::verifyLiteralUpperBits(
+    const spv_parsed_operand_t& parsed_operand, uint32_t word) {
+  // Verify the upper bits of literal numbers not multiple of 32 bits
+  auto verify = [&](uint32_t mask, std::vector<uint32_t> valid_patterns)
+      -> spv_result_t {
+    uint32_t upper = word & mask;
+    for (uint32_t valid : valid_patterns)
+        if (upper == valid)
+          return SPV_SUCCESS;
+    return diagnostic() << "Upper bits of literal number are invalid";
+  };
+
+  switch (parsed_operand.number_kind) {
+    case SPV_NUMBER_SIGNED_INT:
+      switch (parsed_operand.number_bit_width) {
+        case 8:  return verify(0xFFFFFF00,{0x00000000,0xFFFFFF00});
+        case 16: return verify(0xFFFF0000,{0x00000000,0xFFFF0000});
```

Comment:
Question out of sheer ignorance.  There are no 32 bit integers?

Reply:
This check only targets those literal numbers smaller than the word size, i.e. smaller than 32 bits. Therefore, any 32 bit and 64 bit literals pass this test automatically, through the default branch. Thanks for the comments!